### DISCUSSION TITLE
Is there a solution for Markdown Editor?

### DISCUSSION THREAD

**Discussion by xiuno** [slint-ui/slint#5278] [General]
If there is a relevant library, please recommend it or how to implement it by myself. Currently, the idea I can think of is to convert the markdown tag to slint code and render it through slint_interpreter. However, updating is a problem. Do I have to re render the entire window every time? Can we only render the parts without window title and position?

**Comment by Hammerklavier-cn** [comment]
I only know that I can set the style of text for all the words in a single, say, Text element, but how can I apply the setting to part of the text?

**Reply by ogoffart** [reply]
Right now we don't support richtext yet.

Some discussion about richtext in https://github.com/slint-ui/slint/issues/2723  (this talks about editor, but this is also valid for just displaying)

What is currently possible is doing something like that:

```slint
for t in texts : Text {
   text: t.text;
   color: t.color;
   ...
} 
```

Not ideal but might help in some level before we have richtext support.
